### ISSUE TITLE
Don't lint trivially_copy_pass_by_ref on #[inline(always)] functions

### ISSUE THREAD

**Issue by alecmocatta** [rust-lang/rust-clippy#5876]
The suggestion here that `x` "would be more efficient if passed by value" doesn't make much sense to me on an `#[inline(always)]` function. Might be worth only linting if the function isn't marked `#[inline(always)]`?

```rust
#[warn(clippy::pedantic)]

#[inline(always)]
fn foo(x: &i32) {
    println!("{}", x);
}

```

([Playground](https://play.rust-lang.org/?edition=2018&gist=0ab343dc4ab6c78be832510a3f9aae56))

Errors:

```
warning: this argument (4 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)
 --> src/lib.rs:4:11
  |
4 | fn foo(x: &i32) {
  |           ^^^^ help: consider passing by value instead: `i32`
  |
note: the lint level is defined here
 --> src/lib.rs:1:8
  |
1 | #[warn(clippy::pedantic)]
  |        ^^^^^^^^^^^^^^^^
  = note: `#[warn(clippy::trivially_copy_pass_by_ref)]` implied by `#[warn(clippy::pedantic)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref
```
